IdentityCore/src/util/NSDictionary+MSIDExtensions.m

```diff
@@ -59,7 +59,7 @@ + (NSDictionary *)msidDictionaryFromQueryString:(NSString *)string
             continue;
         }
         
-        NSString *key = decode ? [queryElements[0] msidTrimmedString].msidWWWFormURLDecode : [queryElements[0] msidTrimmedString];
+        NSString *key = decode ? [queryElements[0] msidTrimmedString].msidWWWFormURLDecode : [queryElements[0] msidTrimmedString].msidURLDecode;
```

Comment:
Will the caller expect key to be url decoded if they pass decode:false?

Reply:
Yes, this is a private method. 
Calling methods are msidDictionaryFromURLEncodedString and msidDictionaryFromWWWFormURLEncodedString.

Reply:
Both calling methods (msidDictionaryFromWWWFormURLEncodedString and msidDictionaryFromURLEncodedString) expects key and value to be encoded.

Just to be clear, I will rename for better clarity

Reply:
msidDictionaryURLEncodedString: isFormEncoded

Reply:
+ (NSDictionary *)msidDictionaryFromURLEncodedString:(NSString *)string
                                       isFormEncoded:(BOOL)isFormEncoded